src/test/gdb_qpasssignals.py

```diff
@@ -0,0 +1,26 @@
+from turtle import st
+from util import *
+import time
+
+def runTest(passSignalsEnabled: bool):
+    send_gdb(f"set remote pass-signals {'auto' if passSignalsEnabled else 'off'}")
+    send_gdb("handle SIGURG nostop noprint pass")
+
+    start = time.time()
+    for _ in range(30):
+        send_gdb("c")
+
+        expect_rr("XIT-END")
+
+        send_gdb("r")
+
+    return time.time() - start
+
+
+disabledTime = runTest(False)
+enabledTime = runTest(True)
+
+if enabledTime < disabledTime - 5:
+    ok()
+else:
+    failed("Expected larger performance improvement by enabling QPassSignals, this may be a false positive")
```

Comment:
Rather than doing this, how about raising the number of signals to 10,000 or more and just testing that a single reverse-continue from end to start a) is reasonably fast with `QPassSignals` and b) times out without `QPassSignals`. We don't need to test the non-`QPassSignals` state at all.

Reply:
The issue I was running into was that different variants of the test would make a large enough impact on runtime that finding a value for all tests to pass with it enabled and all tests to fail when disabled had a small enough tolerance that it might cause false positives on different machines. The 64 bit variant without the syscall buffer takes the longest, while the 32 bit variant takes the least amount of time. So causing the 32 bit variant to timeout without QPassSignals also caused the 64 bit no syscall buffer variant to timeout with QPassSignals.